src/researches/passiveVoice/morphological/determinePassiveSentence.js

```diff
@@ -0,0 +1,49 @@
+const filter = require( "lodash/filter" );
+const getWords = require( "../../../stringProcessing/getWords.js" );
+
+// Verb-form lists per language
+const getPassiveVerbsRussian = require( "../../russian/passiveVoice/participlesShortenedList.js" )().all;
+
+/**
+ * Matches the sentence against passive verbs.
+ *
+ * @param {string} sentence The sentence to match against.
+ * @param {Array} passiveVerbs The array containing passive verb-forms.
+ * @returns {Array} The found passive verbs.
+ */
+let matchPassiveVerbs = function( sentence, passiveVerbs ) {
+	return filter( passiveVerbs, function( word ) {
```

Comment:
I wonder if matching the other way around is faster. The worst case should be the same (sentenceLength * listLength), but the best case could be so much better (first word in the list is found). Might not matter though. We should benchmark this.

Reply:
I tried both ways on a bunch of Russian texts (short, medium, and long). Surprisingly, the version you suggested takes on average longer to run on all texts. For now, I am keeping the script the way it is.

Reply:
I tried both ways on a bunch of Russian texts (short, medium, and long). The version you suggested is indeed more efficient (based on min time, max time, and average time of performance).